### PR TITLE
ci: update publish workflows, removing old workflow, tweaking main workflow for usability, and updating release docs

### PR DESCRIPTION
**Pull request by aaronsteers**
## Summary by CodeRabbit

- **Bug Fixes**
	- Removed outdated workflow for publishing the SDM connector to DockerHub.
  
- **New Features**
	- Updated version input handling and descriptions in the PyPI publish workflow for better clarity.
	- Changed default setting for `update_connector_builder` input to `false` in the PyPI publish workflow.
  
- **Documentation**
	- Enhanced clarity and scope of the RELEASES.md document, including updates to section titles and simplified instructions for publishing stable and pre-release versions.

These changes improve the clarity and functionality of the publishing workflows, ensuring a smoother experience for users managing package releases.

### CODE REVIEW

File: .github/workflows/pypi_publish.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ on:
         description: "Update Connector Builder. If true, the workflow will create a PR to bump the CDK version used by Connector Builder."
         type: boolean
         required: true
-        default: true
+        default: false
```
Comment (aaronsteers) [inline .github/workflows/pypi_publish.yml:33]: I changed this to default to "false" after I accidentally left it selected for my pre-release test this morning.

Comment (ChristoGrab) [reply .github/workflows/pypi_publish.yml:33]: Good catch!

File: .github/workflows/pypi_publish.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ on:
   workflow_dispatch:
     inputs:
       version:
-        description: "Version. The version to publish, ie 1.0.0 or 1.0.0-dev1. In most cases, you can leave this blank. If run from a release tag (recommended), the version number will be inferred from the git tag."
+        description: "Note that this workflow is intended for prereleases. For public-facing stable releases, please use the GitHub Releases workflow instead: https://github.com/airbytehq/airbyte-python-cdk/blob/main/docs/RELEASES.md. If running this workflow from main or from a dev branch, please enter the desired version number here, for instance 1.2.3dev0 or 1.2.3rc1."
```
Comment (ChristoGrab) [inline .github/workflows/pypi_publish.yml:17]: Love the added context 💪

File: .github/workflows/publish_sdm_connector.yml
Comment (ChristoGrab) [inline .github/workflows/publish_sdm_connector.yml:1]: RIP publish_sdm_connector 🫡

**Review by ChristoGrab — APPROVED** [review]
LGTM, thanks for the keeping the docs so up-to-date!

**Pull request merged**